docs/specifications/tests/Consistency-TP/consistency05.md

```diff
@@ -1,49 +1,209 @@
+
+5. If *IB NS Name Set* is non-empty, then for each name server name in
+   that set do:
+
+   1. Create one A query and one AAAA query with the RD flag unset
+      and name server name as owner name.
+
+   2. For each name server in *NS IP* and for each record 
+      types (A, AAAA):
+      1. Send the address query to the name server.
+      2. If there is no DNS response from the server, then
+         output *[NO_RESPONSE]*.
+      3. Or, if the response is a delegation (referral) to a 
+         sub-zone of *Child Zone*, then:
+         1. Copy the adress query (A, AAAA) that gave the referral
+            response.
+         2. Set the RD flag in the copied query (from unset to set).
+         3. Do a DNS lookup of the the query using a resolving name server 
+            on the public DNS.
+            * The lookup must take into account changes that
+              undelegated data has created, if any.
+         4. If the lookup returns the relevant address record or records,
+            A for A record query and AAAA for AAAA record query, and 
+            with the same owner name as in the query, then extract those 
+            and add to *Address Records From Child* with name and IP 
+            address or addresses.
+      4. Or, if the response has the AA flag unset, then
+         output *[CHILD_NS_FAILED]*. 
+      5. Or, if the RCODE of the response is neither NOERROR nor 
+         NXDOMAIN, then output *[CHILD_NS_FAILED]*.
+      6. Or, if the RCODE is NOERROR (with the AA flag set), then
+         extract any address records (A, AAAA) from the answer
+         section whose owner name matches the owner name 
+         of the query and add that or those to 
+         *Address Records From Child* with name and IP.
+      7. Else, there is nothing to do (i.e. RCODE is NXDOMAIN).
+
+   3. If all servers outputted *[NO_RESPONSE]* or *[CHILD_NS_FAILED]*, 
+      then output *[CHILD_ZONE_LAME]* and completely stop processing 
+      this test case.
+
+6. Compare the IP address for the name servers from 
+   *Delegation Strict Glue* with *Address Records From Child*
+   (i.e. [in-bailiwick] only).
+
+   1. If an IP from *Delegation Strict Glue* is not listed in 
+      *Address Records From Child* with that same name server name, 
+      then output *[IN_BAILIWICK_ADDR_MISMATCH]*.
+
+   2. If an IP from *Address Records From Child* is not listed in
+      *Delegation Strict Glue* with that same name server name, then 
+      output *[EXTRA_ADDRESS_CHILD]*.
+
+7. For each name server name in *Delegation Extended Glue* 
+   (i.e. [out-of-bailiwick] only) ("DEG Name Server Name") do: 
+
+   1. Do two DNS lookups, one record type A and one record type 
+      AAAA, for *DEG Name Server Name* on public DNS and create a
+      set of the IP addresses from the A an AAAA records, respectively,
+      from the answer sections of the responses. Do not follow any CNAME. 
```

Comment:
This is the only query with an instruction not to follow CNAMEs. Is it implied that you should follow CNAMEs in the other queries?

Reply:
No, CNAME should not be followed in this test case. In the other two possible cases there was an indirect restriction through owner name. I have update all three to have the same format. (Updated in a separate commit.)